modules/util/source-formatter/src/main/java/com/liferay/source/formatter/util/SourceFormatterUtil.java

```diff
@@ -552,11 +661,76 @@ private static PathMatchers _getPathMatchers(
 		return pathMatchers;
 	}
 
+	private static void _populateIgnoreDirectories(String baseDirName) {
+		_sfIgnoreDirectories = new ArrayList<>();
+		_subrepoIgnoreDirectories = new ArrayList<>();
+
+		git(
+			Arrays.asList(
+				"ls-files", "--", "**/source_formatter.ignore", "**/.gitrepo"),
+			baseDirName, null, false,
+			filePath -> {
+				filePath = filePath.replace(
+					StringPool.BACK_SLASH, StringPool.SLASH);
+
+				if (filePath.endsWith("/source_formatter.ignore")) {
+					File file = new File(baseDirName, filePath);
+
+					_sfIgnoreDirectories.add(file.getParent());
+				}
+
+				if (filePath.endsWith("/.gitrepo")) {
+					String content = null;
+
+					File file = new File(baseDirName, filePath);
+
+					try {
+						content = FileUtil.read(file);
+					}
+					catch (IOException ioException) {
+						throw new RuntimeException(ioException);
+					}
+
+					if (content.contains("autopull = true")) {
+						_subrepoIgnoreDirectories.add(file.getParent());
+					}
+				}
+			});
+	}
+
 	private static List<String> _scanForFiles(
 			final String baseDirName, final PathMatchers pathMatchers,
 			final boolean includeSubrepositories)
 		throws IOException {
 
+		try {
+			if (GitUtil.getLatestCommitId() != null) {
+				if ((_sfIgnoreDirectories == null) ||
+					(_subrepoIgnoreDirectories == null)) {
+
+					_populateIgnoreDirectories(baseDirName);
+				}
+
+				List<String> gitFiles = new ArrayList<>();
+
+				git(
+					Arrays.asList("ls-files"), baseDirName, pathMatchers,
+					includeSubrepositories,
+					line -> gitFiles.add(
+						StringBundler.concat(
+							baseDirName, StringPool.FORWARD_SLASH,
```

Comment:
Redundant `StringPool.FORWARD_SLASH`
@drewbrokke Can you help remove it? but this PR is merged in upstream yet.

Reply:
/cc @petershinn